### PR TITLE
Migrate retry to null-safety

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

File: retry/lib/retry.dart [hunk 1]
```diff
@@ -120,13 +120,13 @@ class RetryOptions {
...
-    FutureOr<void> Function(Exception) onRetry,
+    FutureOr<bool> Function(Exception)? retryIf,
+    FutureOr<void> Function(Exception)? onRetry,
   }) async {
-    int attempt = 0;
+    var attempt = 0;
     // ignore: literal_only_boolean_expressions
     while (true) {
-      attempt++; // first invocation is the first attempt
+      attempt++; // first invocation is the first attempt†
```
Comment (jonasfj) [inline retry/lib/retry.dart:129]: What is up with `†` ?

File: retry/pubspec.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name: retry
-version: 3.0.1
+version: 3.0.1-null-safety
```
Comment (jonasfj) [inline retry/pubspec.yaml:2]:
```suggestion
version: 3.0.1-null-safety.0
```

File: retry/pubspec.yaml [hunk 3]
```diff
@@ -10,4 +10,4 @@ dev_dependencies:
   test: ^1.5.1
   pedantic: ^1.4.0
 environment:
-  sdk: ">=2.0.0 <3.0.0"
+  sdk: ">=2.12.0-0.0.dev <3.0.0"
```
Comment (jonasfj) [inline retry/pubspec.yaml:13]:
```suggestion
  sdk: ">=2.12.0-0 <2.12.0"
```

**Review by jonasfj — CHANGES_REQUESTED** [review]

File: retry/lib/retry.dart (hunk 1 above)
Comment (sigurdm) [reply retry/lib/retry.dart:129]: Hmm - no clue where that came from...

Comment (sigurdm) [reply retry/lib/retry.dart]: Removed

**Review by jonasfj — APPROVED** [review]

File: retry/pubspec.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name: retry
-version: 3.0.1
+version: 3.0.1-null-safety.0
```
Comment (natebosch) [inline retry/pubspec.yaml:2]: This should be `4.0.0-nullsafety`

- Packages that aren't pinned by flutter should use a breaking change.
- I think there can't be a `-` in the portion of the version following the first `-`. @jakemac53 do you remember if that is the case?
- The existing published `3.0.1` takes precedence over this one.

**Pull request merged**